### ISSUE TITLE
Hyphenation not working when a nbsp is present elsewhere in the word

### ISSUE THREAD

**Issue by ondras** [Kozea/WeasyPrint#2270]
I am not sure whether this is a bug or feature. See this rendering:

![image](https://github.com/user-attachments/assets/3ace4b2c-88c8-4507-9cb3-a64a57b6798e)

The "x komunikace" is not hyphenated when the space is non-breakable. I would say that it is a bug, because browsers line-break both cases in the same fashion.

**Comment by ondras** [comment]
I might have fixed this in a PR: https://github.com/Kozea/WeasyPrint/pull/2271

(I have absolutely zero experience with Pango and the underlying line-breaking algorithm in `line_break.py` is very complex, so please double-check my code.)

**Comment by liZe** [comment]
Here’s what I get with various browsers. Firefox seems to be the only one to do things right.

Chrome:
![Image](https://github.com/user-attachments/assets/d3e9f1f9-ee28-4935-991a-45182ae5f6b7)

Firefox:
![Image](https://github.com/user-attachments/assets/0f6a6735-0176-4180-be65-b735c67cfefc)

Gnome Web:
![Image](https://github.com/user-attachments/assets/df07b728-995d-4816-9c47-7ebc558f4a60)

Here’s what I get with WeasyPrint:
![Image](https://github.com/user-attachments/assets/39872a0f-4b8c-44d2-b603-cad20c0ad4ef)

WeasyPrint can break the first word correctly, but not the second one.

**Comment by ondras** [comment]
Right. The issue is visible when the hyphenation opportunity is in the second word (of the nbsp-joined word pair). This is very typical for Czech typography, where we join a one-letter word with the (longer) following word using a nbsp (as seen in my first post).
